sci-mathematics/frobby/frobby-0.9.0-r2.ebuild

```diff
@@ -1,31 +1,35 @@
-# Copyright 1999-2013 Gentoo Foundation
+# Copyright 1999-2017 Gentoo Foundation
 # Distributed under the terms of the GNU General Public License v2
 # $Id$
 
-EAPI=4
+EAPI=6
 
-inherit eutils
+inherit toolchain-funcs
 
 DESCRIPTION="Software system and project for computations with monomial ideals"
 HOMEPAGE="http://www.broune.com/frobby/"
 SRC_URI="http://www.broune.com/frobby/frobby_v${PV}.tar.gz"
 
 LICENSE="GPL-2"
 SLOT="0"
-KEYWORDS="amd64 ~arm x86"
+KEYWORDS="~amd64 ~arm ~x86"
 IUSE="doc static-libs"
 
-RDEPEND="dev-libs/gmp[cxx]"
+RDEPEND="dev-libs/gmp:0=[cxx]"
 DEPEND="${RDEPEND}
 	doc? ( virtual/latex-base )"
 
 S="${WORKDIR}/frobby_v${PV}"
 
+PATCHES=(
+	"${FILESDIR}/${PN}-cflags-no-strip-soname.patch"
+	"${FILESDIR}/${PN}-gcc-4.7.patch"
+	"${FILESDIR}/${PN}-gmp-5.1.patch"
+)
+
 src_prepare() {
-	epatch \
-		"${FILESDIR}/${PN}-cflags-no-strip-soname.patch" \
-		"${FILESDIR}/${PN}-gcc-4.7.patch" \
-		"${FILESDIR}/${PN}-gmp-5.1.patch"
+	default
+
 	# CXXFLAGS are called CPPFLAGS
 	sed "s/CPPFLAGS/CXXFLAGS/" -i Makefile || die
 }
```

Comment:
can you try `tc-export CXX` and then removing all the `CXX=` specifiers in `src_compile`?